code/modules/power/smes.dm

```diff
@@ -221,18 +227,60 @@
 
 /obj/machinery/power/smes/update_overlays()
 	. = ..()
-	if(machine_stat & BROKEN)
-		return
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
 
 	if(panel_open)
+		. += "panel"
 		return
 
-	. += "smes-op[outputting ? 1 : 0]"
-	. += "smes-oc[inputting ? 1 : 0]"
+	if(machine_stat & BROKEN)
+		return
 
 	var/clevel = chargedisplay()
-	if(clevel > 0)
-		. += "smes-og[clevel]"
+	if(!terminal)
+		. += "failing"
+		SSvis_overlays.add_vis_overlay(src, icon, "failing", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "failing", layer, EMISSIVE_PLANE, dir)
+	if(clevel>0)
+		. += "charge[clevel]"
+		SSvis_overlays.add_vis_overlay(src, icon, "charge[clevel]", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "charge[clevel]", layer, EMISSIVE_PLANE, dir)
+	if(is_emped)
+		. += "emp"
+		SSvis_overlays.add_vis_overlay(src, icon, "emp", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "emp", layer, EMISSIVE_PLANE, dir)
+	else
+		if(inputting)
+			if(clevel == SMES_CLEVEL_5)
+				. += "input-2"
+				SSvis_overlays.add_vis_overlay(src, icon, "input-2", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "input-2", layer, EMISSIVE_PLANE, dir)
+			else
+				. += "input-1"
+				SSvis_overlays.add_vis_overlay(src, icon, "input-1", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "input-1", layer, EMISSIVE_PLANE, dir)
+		else if(input_attempt)
+			. += "input-0"
+			SSvis_overlays.add_vis_overlay(src, icon, "input-0", layer, plane, dir)
+			SSvis_overlays.add_vis_overlay(src, icon, "input-0", layer, EMISSIVE_PLANE, dir)
+		else
+			. += "input-off"
+			SSvis_overlays.add_vis_overlay(src, icon, "input-off", layer, plane, dir)
+			SSvis_overlays.add_vis_overlay(src, icon, "input-off", layer, EMISSIVE_PLANE, dir)
+
+		if(outputting)
+			if(clevel == SMES_CLEVEL_5)
+				. += "output2"
+				SSvis_overlays.add_vis_overlay(src, icon, "output2", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "output2", layer, EMISSIVE_PLANE, dir)
+			else
+				. += "output1"
+				SSvis_overlays.add_vis_overlay(src, icon, "output1", layer, plane, dir)
+				SSvis_overlays.add_vis_overlay(src, icon, "output1", layer, EMISSIVE_PLANE, dir)
+		else
+			. += "output0"
+			SSvis_overlays.add_vis_overlay(src, icon, "output0", layer, plane, dir)
+			SSvis_overlays.add_vis_overlay(src, icon, "output0", layer, EMISSIVE_PLANE, dir)
```

Comment:
use early returns <https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns> at least so far as is possible (likely will only work with the uppermost block)

Reply:
Anywhere else than under EMP state, it wont work and block overlays.